### ISSUE TITLE
Error Report - Request for Permission

### ISSUE THREAD

**Issue by Yikerdoodle** [checkra1n/BugTracker#351]
**What device + iOS version are you on?**
iPhone 7 Plus, iOS 13.1.3

**What checkra1n version are you using?**
Was on 0.9.2 working perfectly, updated to 0.9.3.2 and this error showed.

**What are the steps to reproduce the issue?**
I don’t know.
I just tried to update/rejailbreak with computer and this error showed.

**What do you expect, and what is happening instead?**
I expected to be successfully rejailbroken, but this error showed up saying
“Error Report - Request for Permission...” (Rest is in Screenshot)

**Any other info, error logs, screenshots, ...?**
After trying with 0.9.3.2, I tried with 0.9.3 and 0.9.2 multiple times as well. Same error still showed up. 0.9.2 worked perfectly before. Also tried restoring system through Safe Mode, then jailbreaking and error still shows up. Tried removing half of tweaks through Safe Mode, but no change. Using MacOS High Sierra, not a VM/Hackintosh.
![5DFDF7D9-DDE8-4F7E-9B43-D071A5595D7A](https://user-images.githubusercontent.com/47406347/68879034-56f2be00-06ce-11ea-85b6-ad84144e95b8.jpeg)

**Comment by CrazyKidd** [comment]
0.9.3 works perfectly on my iPhone with iOS 12.4.1, but 0.9.3.2 did not work, beside when I successfully rejailbroken device with 0.9.3, checkra1n app disappears from my iPhone, how strange.

**Comment by Yikerdoodle** [comment]
> 0.9.3 works perfectly on my iPhone with iOS 12.4.1, but 0.9.3.2 did not work, beside when I successfully rejailbroken device with 0.9.3, checkra1n app disappears from my iPhone, how strange.

That is strange. Hopefully the devs see this, so they can fix it.

**Comment by CrazyKidd** [comment]
You can use cmd "./checkra1n.app/Contents/MacOS/checkra1n_gui -" from console to get more error details or try to use other old versions of checkra1n.

**Comment by Yikerdoodle** [comment]
I’ve already tried on all versions except the initial release.

**Comment by kpwn** [comment]
Please try https://assets.checkra.in/test/checkra1n%20beta%200.9.3.7.dmg

**Comment by youran1314** [comment]
> Please try https://assets.checkra.in/test/checkra1n%20beta%200.9.3.7.dmg

nope still doesnt work 
still displays this message "Setting up the exploit (this is the heap spray)"

**Comment by johnwatson122** [comment]
@NickKrahn 
Have you updated your itunes or drivers? It worked for me after doing so

**Comment by kpwn** [comment]
@youran1314 sounds like your USB controller or drivers are not capable of performing checkm8. what model of mac is this? have you tried rebooting the computer?

**Comment by youran1314** [comment]
@kpwn its the 24" 2008 imac, i restarted the computer multiple times still couldnt get it to work

**Comment by kpwn** [comment]
ah hmm, that's pretty ancient, it *should* work, but it's possible the USB stack is different enough as to make it not work. you've said older versions worked for you however, correct? that's pretty odd, we shouldn't have changed much in the usb code between releases yet.

**Comment by youran1314** [comment]
@kpwn 
oh, no lol that's the other person saying the older version worked for him

**Comment by kpwn** [comment]
oh hah, my bad. it's possible your controller is just unsupported :\

**Comment by youran1314** [comment]
@kpwn will it be supported in the future? or am i just doomed

**Comment by kpwn** [comment]
it's hard to say, i'd need to be able to figure out exactly what's not working, and it's hard not having a sample on hand.. unless it's just due to the OS version, in which case it's possible some minor changes could fix it. it's low priority however..

**Comment by Yikerdoodle** [comment]
> Please try https://assets.checkra.in/test/checkra1n%20beta%200.9.3.7.dmg

@kpwn I will definitely try that when I have time today.

**Comment by youran1314** [comment]
@kpwn just to let you know. I got the same error as this guy https://github.com/checkra1n/BugTracker/issues/255 so you might be able to get a clue about what's the problem

**Comment by kpwn** [comment]
ah huh, you have 10.10 in common, so it might be an OS specific thing. i will try to set up a 10.10 vm to test it later.

**Comment by Yikerdoodle** [comment]
@johnwatson122 my firmware is the latest High Sierra and the latest my iMac (mid-2010) can get. I’m pretty sure my iTunes is also the latest.

**Comment by Yikerdoodle** [comment]
@kpwn tried that dmg link. Shows same error message. Though I just realized that there is another error on the app itself.
![C9C1CF6A-3CE0-4BB0-8686-E2007F9640EF](https://user-images.githubusercontent.com/47406347/68899755-f5dfe000-06f7-11ea-95cd-52d1cc6160e7.jpeg)

**Comment by kpwn** [comment]
Mind giving https://assets.checkra.in/test/checkra1n%20beta%200.9.4.dmg a shot? Not sure if it will help but it might.

**Comment by Yikerdoodle** [comment]
@kpwn I’ll try it out as soon as I have access to a computer.

**Comment by Yikerdoodle** [comment]
It worked now because after I restored the system, I initially booted into Safe Mode. Then I went into Cydia and updated Bourne again shell which was in upgrades section. Then I booted into jailbroken mode and it worked. Hmmm. Was it not working the other times because Bourne again shell wasn’t updated?

**Comment by sSiDs** [comment]
i have hackintosh 10.10.5.....and got -31 error on my 5S device. Do I need a real MAC?

**Comment by johnwatson122** [comment]
Can you update?

**Comment by schmurtzm** [comment]
I have a VMware mac OSX (10.15.1) and I've got the same message  : error report- request for persimission -> "Error code: -31"

![image](https://user-images.githubusercontent.com/7110113/69962903-500ddd00-150f-11ea-822d-93094b074240.png)


I tried USB vm settings in 3.0 ,2.0 and 1.1 (it's better in USB 2.0 than usb 3.0 for the iphone detection) but without success.

May be it could help to launch the jailbreak only when you press a button to be able to disconnect/reconnect the usb port on the iphone when you're in DFU mode before starting....

**Comment by Yikerdoodle** [comment]
Hey everyone, so I solved my issue by booting into safe mode and updating all pending tweaks and now it works to boot into jailbroken state. Sorry, I haven’t been on GitHub for a while and I will now close this issue.

**Comment by mtchis** [comment]
> Tôi có VMware mac OSX (10.15.1) và tôi đã nhận được thông báo tương tự: báo cáo lỗi- yêu cầu thuyết phục -> "Mã lỗi: -31"
> 
> ![hình ảnh](https://user-images.githubusercontent.com/7110113/69962903-500ddd00-150f-11ea-822d-93094b074240.png
> 
> Tôi đã thử cài đặt USB vm ở 3.0, 2.0 và 1.1 (trong USB 2.0 tốt hơn USB 3.0 để phát hiện iphone) nhưng không thành công.
> 
> Có thể chỉ có thể khởi chạy jailbreak khi bạn nhấn nút để có thể ngắt kết nối / kết nối lại cổng usb trên iphone khi bạn ở chế độ DFU trước khi bắt đầu ....

hello do you repaired this ?
